src/app/add_entry/page.jsx

```diff
@@ -0,0 +1,157 @@
+"use client"
+import Link from 'next/link'
+import { supabase } from '../lib/supabase'
+import { useRouter } from 'next/navigation'
+import { UploadButton } from '../utils/uploadthing'
+import { useState } from 'react'
+
+const weatherOptions = [
+  "sunny",
+  "raining",
+  "cloudy"
+]
+
+function AddEntry() {
+  const router = useRouter()
+
+  const [uploadedFileUrls, setUploadedFileUrls] = useState()
+  const [submitDisabled, setSubmitDisabled] = useState(false)
+
+  const handleSubmit = async (e) => {
+    // Prevent page from refreshing on Submit click
+    e.preventDefault()
+
+    const formData = new FormData(e.target);
+    const formProps = Object.fromEntries(formData);
+
+    const submitData = async () => {
+      try {
+        const { error } = await supabase
+          .from('diary_entries')
+          .insert({ date: formProps.date, description: formProps.description, weather: formProps.weather, images: uploadedFileUrls })
+        if (error) {
+          throw error;
+        }
+        alert('Saved!')
+        // Redirect User to List screen after saving
+        router.push('/')
+      } catch (error) {
+        console.error('Error fetching data from Supabase:', error.message);
+      }
+    };
+    submitData();
+  }
+
+  return (
+    <>
+      <form onSubmit={handleSubmit}>
+        <div className="space-y-12">
+          <div className="border-b border-gray-900/10 pb-12">
+            <div className="grid grid-cols-1 gap-x-6 gap-y-6 sm:grid-cols-12">
+              {/* date */}
+              <div className="sm:col-span-3">
+                <label htmlFor="date" className="block text-sm font-medium leading-6 text-gray-900">
+                  Date
+                </label>
+                <div className="mt-2">
+                  <input
+                    type="date"
+                    name="date"
+                    id="date"
+                    required
+                    className="block w-full rounded-md py-1 px-2 border-0 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
+                  />
+                </div>
+              </div>
```

Comment:
On one hand, I don't mind the comments above these components, they will be stripped out of a production build anyways. But when possible, it's good for the code to make itself obvious.

If you created a `<DateInput />` and `<WeatherInput />`, etc.. component they would be quite obvious what they are for based on the component name. Plus it's reusable for other pages if needed :)